### PR TITLE
Support crawling area-owners.json if it exists

### PR DESCRIPTION
**Pull request by jeffhandley**
Along with the crawling functionality, this PR includes a utility script that will download an area-owners.md file and produce an area-owners.json file, expanding GitHub team members found to be owners.

### CODE REVIEW

File: src/IssueDb/Crawling/CrawledAreaOwnerFile.cs [hunk 1]
```diff
@@ -21,25 +23,51 @@ public CrawledAreaOwnerFile(IEnumerable<CrawledAreaOwnerEntry> entries)
...
             while (maxTries-- > 0)
             {
                 var client = new HttpClient();
-                try
-                {
-                    var contents = await client.GetStringAsync(url);
-                    return Parse(contents);
-                }
-                catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+
```
Comment (jeffhandley) [inline src/IssueDb/Crawling/CrawledAreaOwnerFile.cs:33]: I recommend reviewing with whitespace differences collapsed.
https://github.com/terrajobst/issuesof.net/pull/12/files?w=1

File: src/IssueDb/Crawling/CrawledAreaOwnerFile.cs [hunk 2]
```diff
@@ -21,25 +23,51 @@ public CrawledAreaOwnerFile(IEnumerable<CrawledAreaOwnerEntry> entries)
...
+                if (!jsonFileNotFound)
                 {
-                    // In this case we know that the file doesn't exist.
-                    return null;
+                    try
+                    {
+                        // First attempt to load a JSON file
+                        var jsonEntries = await client.GetFromJsonAsync<CrawledAreaOwnerJsonFile>(jsonUrl);
+                        return new CrawledAreaOwnerFile(jsonEntries.Areas.Select(area => new CrawledAreaOwnerEntry(area.Label, area.Lead, area.Owners)));
```
Comment (jeffhandley) [inline src/IssueDb/Crawling/CrawledAreaOwnerFile.cs:40]:
```suggestion
                        return new CrawledAreaOwnerFile(jsonEntries.Areas.Select(area => new CrawledAreaOwnerEntry(area.Label.Replace("area-", ""), area.Lead, area.Owners)));
```

**Comment by eiriktsarpalis** [comment]
Was about to open a similar PR when I stumbled on this one. We really need this for endgame, anything I can do to get this merged soon?

File: scripts/area-owners-md-to-json.ps1 [hunk 3]
```diff
@@ -0,0 +1,45 @@
+$owner = "dotnet"
```
Comment (terrajobst) [inline scripts/area-owners-md-to-json.ps1:1]: The intend is that this is only used offline when we're migrating, right?

**Review by terrajobst — APPROVED** [review]

File: scripts/area-owners-md-to-json.ps1 (hunk 3 above)
Comment (jeffhandley) [reply scripts/area-owners-md-to-json.ps1:1]: Right. We might reuse the script, but it'll be run manually each time.

**Pull request merged**